src/tcgwars/logic/impl/gen8/RebelClash.groovy

```diff
@@ -3771,7 +3771,7 @@ public enum RebelClash implements LogicCardInfo {
               }
             }
             after APPLY_ATTACK_DAMAGES, {
-              if(attackDidDamage && self.cards.contains(thisCard)) { // this energy card is still attached
+              if(attackDidDamage && self.cards.contains(thisCard) && ef.attacker) { // this energy card is still attached and the attacking Pokemon is still in play
```

Comment:
this `ef.attacker` condition will always be true, thus have no effect. are you trying to check whether the pokemon in question is still in the game? if yes, then it's better to use `ef.attacker.inPlay` instead

Reply:
Thanks for the suggestion, The bug that I was trying to fix was a null pointer exception on attacks that shuffle away the attacking pokemon. I would guess that ef.attacker.inPlay might throw the same exception if ef.attacker is null?